### ISSUE TITLE
Unit test failing in CI: OtlpHttpExporterTestPeer.ExportJsonIntegrationTest

### ISSUE THREAD

**Issue by lalitb** [open-telemetry/opentelemetry-cpp#1049]
One of the failure instances:
https://github.com/open-telemetry/opentelemetry-cpp/runs/4083453037?check_suite_focus=true

```
[ RUN      ] OtlpHttpExporterTestPeer.ExportJsonIntegrationTest
exporters/otlp/test/otlp_http_exporter_test.cc:240: Failure
Value of: waitForRequests(2, old_count + 1)
  Actual: false
Expected: true
[  FAILED  ] OtlpHttpExporterTestPeer.ExportJsonIntegrationTest (4446 ms)
```

**Comment by ThomsonTan** [comment]
Wondering why this was not caught by our PR build.

**Comment by lalitb** [comment]
> Wondering why this was not caught by our PR build.

I don't think it is failing consistently, so not got caught in our PR build

**Comment by esigo** [comment]
This was happening here as well https://github.com/open-telemetry/opentelemetry-cpp/pull/1044#issuecomment-955698045.

**Comment by lalitb** [comment]
Haven't seen this happening this week, probably a transient issue with the CI pipeline. Will watch for some more time before closing.

**Comment by owent** [comment]
I find it take a lot time to finish tests when run under `valgrind` . It almost take 6 seconds to finish this test even in a successful case. 
Maybe just increasing timeout could help?

**Comment by esigo** [comment]
It happened [again](https://github.com/open-telemetry/opentelemetry-cpp/runs/4955500860?check_suite_focus=true#step:5:134). I'll try to raise a PR with mock in the next coming days.

**Comment by owent** [comment]
> It happened [again](https://github.com/open-telemetry/opentelemetry-cpp/runs/4955500860?check_suite_focus=true#step:5:134). I'll try to raise a PR with mock in the next coming days.

Thanks. Maybe it's also better to use multi_handle instead of start a new thread to start a asynchronous HTTP request when using libcurl.Should I start another issue?

**Comment by lalitb** [comment]
> Thanks. Maybe it's also better to use multi_handle instead of start a new thread to start a asynchronous HTTP request when using libcurl.Should I start another issue?

Thank @owent. Which thread invocation are we talking about here? The only place a potential thread may get created internally for libcurl is here -
https://github.com/open-telemetry/opentelemetry-cpp/blob/16a9c53680079a7a5f6728dd4bf4938efa214b30/ext/include/opentelemetry/ext/http/client/curl/http_operation_curl.h#L298

Do we want to remove it, and instead use multi_handle?

**Comment by owent** [comment]
> Reference in



> > Thanks. Maybe it's also better to use multi_handle instead of start a new thread to start a asynchronous HTTP request when using libcurl.Should I start another issue?
> 
> Thank @owent. Which thread invocation are we talking about here? The only place a potential thread may get created internally for libcurl is here -
> 
> https://github.com/open-telemetry/opentelemetry-cpp/blob/16a9c53680079a7a5f6728dd4bf4938efa214b30/ext/include/opentelemetry/ext/http/client/curl/http_operation_curl.h#L298
> 
> Do we want to remove it, and instead use multi_handle?

Yes, I can create another issue and try to use [`multi_handle`](https://curl.se/libcurl/c/libcurl-multi.html) after #1209 is merged.It can reduce the cost when there are a lot sessions.
